diff --git a/dist/components/variant/default/TextInput.svelte b/dist/components/variant/default/TextInput.svelte
index c71e036a5ef9fc1af3beec1890956577c841a229..764c5b382954f33069dedc52677c3dbdee0dab13 100644
--- a/dist/components/variant/default/TextInput.svelte
+++ b/dist/components/variant/default/TextInput.svelte
@@ -94,7 +94,7 @@
 	{#if textInputModes.length > 1}
 		<button type="button" onclick={() => (mode = nextMode)}>
 			<span class="disappear" aria-hidden="true">{texts.color[mode]}</span>
-			<span class="appear">{texts.changeTo} {texts.color[nextMode]}</span>
+			<span class="appear">{texts.changeTo.includes('{mode}') ? texts.changeTo.replace('{mode}', texts.color[nextMode]) : `${texts.changeTo} ${texts.color[nextMode]}`}</span>
 		</button>
 	{:else}
 		<div class="button-like">{texts.color[mode]}</div>
