diff --git a/dist/components/variant/default/TextInput.svelte b/dist/components/variant/default/TextInput.svelte
index 371843460b3c6c76b0a7a7011b493bad17d6153e..55c71b4d3ef886d218715611d7c3551fed318d94 100644
--- a/dist/components/variant/default/TextInput.svelte
+++ b/dist/components/variant/default/TextInput.svelte
@@ -96,7 +96,7 @@
 	{#if textInputModes.length > 1}
 		<button type="button" onclick={() => (mode = nextMode)}>
 			<span class="disappear" aria-hidden="true">{texts.color[mode]}</span>
-			<span class="appear">{texts.changeTo} {texts.color[nextMode]}</span>
+			<span class="appear">{texts.changeTo.includes('{mode}') ? texts.changeTo.replace('{mode}', texts.color[nextMode]) : `${texts.changeTo} ${texts.color[nextMode]}`}</span>
 		</button>
 	{:else}
 		<div class="button-like">{texts.color[mode]}</div>
