From fc67f0ca1a5d2edfa262e69e18e20df4c2d0ea88

diff --git a/LICENSE.MP4 b/LICENSE.MP4
index 30137c3..4187cf0 100644
--- a/LICENSE.MP4
+++ b/LICENSE.MP4
@@ -1,5 +1,5 @@
 This software uses code of FFmpeg <http://ffmpeg.org> licensed under
-the GPLv2 <https://www.gnu.org/licenses/gpl-2.0.html>
+the LGPLv2.1 <http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html>
 and its source can be downloaded here <https://ffmpeg.org/download.html>.
 
 ==============================================================================
@@ -10,9 +10,50 @@ and its source can be downloaded here <https://github.com/rbrito/lame>.
 
 ==============================================================================
 
-This software uses code of x264 licensed under the GPLv2
-<https://www.gnu.org/licenses/gpl-2.0.html> and its source can be
-downloaded here <https://www.videolan.org/developers/x264.html>.
+Copyright 2001-2011 Xiph.Org, Skype Limited, Octasic,
+                    Jean-Marc Valin, Timothy B. Terriberry,
+                    CSIRO, Gregory Maxwell, Mark Borgerding,
+                    Erik de Castro Lopo
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions
+are met:
+
+- Redistributions of source code must retain the above copyright
+notice, this list of conditions and the following disclaimer.
+
+- Redistributions in binary form must reproduce the above copyright
+notice, this list of conditions and the following disclaimer in the
+documentation and/or other materials provided with the distribution.
+
+- Neither the name of Internet Society, IETF or IETF Trust, nor the
+names of specific contributors, may be used to endorse or promote
+products derived from this software without specific prior written
+permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER
+OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+Opus is subject to the royalty-free patent licenses which are
+specified at:
+
+Xiph.Org Foundation:
+https://datatracker.ietf.org/ipr/1524/
+
+Microsoft Corporation:
+https://datatracker.ietf.org/ipr/1914/
+
+Broadcom Corporation:
+https://datatracker.ietf.org/ipr/1526/
 
 ==============================================================================
 
diff --git a/Makefile b/Makefile
index 1b90983..930dadb 100644
--- a/Makefile
+++ b/Makefile
@@ -6,15 +6,9 @@ PRE_JS = build/pre.js
 POST_JS_SYNC = build/post-sync.js
 POST_JS_WORKER = build/post-worker.js
 
-COMMON_FILTERS = aresample scale crop overlay
-COMMON_DEMUXERS = matroska ogg avi mov flv mpegps image2 mp3 concat
-COMMON_DECODERS = \
-	vp8 vp9 theora \
-	mpeg2video mpeg4 h264 hevc \
-	png mjpeg \
-	vorbis opus \
-	mp3 ac3 aac \
-	ass ssa srt webvtt
+COMMON_FILTERS = pan aresample apad atrim amerge
+COMMON_DEMUXERS = ogg
+COMMON_DECODERS = libopus
 
 WEBM_MUXERS = webm ogg null image2
 WEBM_ENCODERS = libvpx_vp8 libopus mjpeg
@@ -34,15 +28,17 @@ WEBM_SHARED_DEPS = \
 	build/opus/dist/lib/libopus.so \
 	build/libvpx/dist/lib/libvpx.so
 
-MP4_MUXERS = mp4 mp3 null
-MP4_ENCODERS = libx264 libmp3lame aac
+# Note: We don't generate iPod files, but the "ipod" muxer is the one that
+# supports MP4+ALAC
+MP4_MUXERS = flac ipod mp3 mp4 ogg wav null
+MP4_ENCODERS = flac aac alac libmp3lame pcm_s16le
 FFMPEG_MP4_BC = build/ffmpeg-mp4/ffmpeg.bc
-FFMPEG_MP4_PC_PATH = ../x264/dist/lib/pkgconfig
+FFMPEG_MP4_PC_PATH = ../opus/dist/lib/pkgconfig
 MP4_SHARED_DEPS = \
 	build/lame/dist/lib/libmp3lame.so \
-	build/x264/dist/lib/libx264.so
+	build/opus/dist/lib/libopus.so
 
-all: webm mp4
+all: ffmpeg-worker-mp4.js
 webm: ffmpeg-webm.js ffmpeg-worker-webm.js
 mp4: ffmpeg-mp4.js ffmpeg-worker-mp4.js
 
@@ -292,9 +288,8 @@ build/ffmpeg-mp4/ffmpeg.bc: $(MP4_SHARED_DEPS)
 		$(FFMPEG_COMMON_ARGS) \
 		$(addprefix --enable-encoder=,$(MP4_ENCODERS)) \
 		$(addprefix --enable-muxer=,$(MP4_MUXERS)) \
-		--enable-gpl \
 		--enable-libmp3lame \
-		--enable-libx264 \
+		--enable-libopus \
 		--extra-cflags="-I../lame/dist/include" \
 		--extra-ldflags="-L../lame/dist/lib" \
 		&& \
@@ -306,7 +301,7 @@ build/ffmpeg-mp4/ffmpeg.bc: $(MP4_SHARED_DEPS)
 # for simple tests and 32M tends to run slower than 64M.
 EMCC_COMMON_ARGS = \
 	--closure 1 \
-	-s TOTAL_MEMORY=67108864 \
+	-s ALLOW_MEMORY_GROWTH=1 \
 	-s OUTLINING_LIMIT=20000 \
 	-O3 --memory-init-file 0 \
 	--pre-js $(PRE_JS) \
